pkg/kubefed2/unjoin.go

```diff
@@ -163,39 +161,19 @@ func UnjoinCluster(hostConfig, clusterConfig *rest.Config, federationNamespace,
 		return err
 	}
 
-	glog.V(2).Infof("Performing preflight checks for unjoin cluster: %s.", unjoiningClusterName)
-	err = performPreflightChecksForUnjoin(clusterClientset, unjoiningClusterName, hostClusterContext, federationNamespace)
-	if err != nil {
-		return err
-	}
-
 	if removeFromRegistry {
-		err = removeFromClusterRegistry(hostConfig, clusterNamespace, clusterConfig.Host, unjoiningClusterName, dryRun)
-		if err != nil {
-			return err
-		}
+		removeFromClusterRegistry(hostConfig, clusterNamespace, clusterConfig.Host, unjoiningClusterName, dryRun)
 	}
 
 	glog.V(2).Infof("Deleting federated cluster resource from namespace: %s for unjoin cluster: %s",
 		federationNamespace, unjoiningClusterName)
 
-	_, err = deleteFederatedClusterAndSecret(hostClientset, fedClientset, federationNamespace,
+	deleteFederatedClusterAndSecret(hostClientset, fedClientset, federationNamespace,
 		unjoiningClusterName, dryRun)
-	if err != nil {
-		glog.V(2).Infof("Failed to delete federated cluster resource from namespace: %s for unjoin cluster: %s due to: %v",
-			federationNamespace, unjoiningClusterName, err)
-		return err
-	}
-
 	glog.V(2).Infof("Deleted federated cluster resource from namespace: %s for unjoin cluster: %s",
```

Comment:
This also should only be logged if there is no error for `deleteFederatedClusterAndSecret`, so how about keep the return value of `deleteFederatedClusterAndSecret` and only log this message `if err == nil`? Same for other functions.